### PR TITLE
Remove codenav error message if status is 0

### PR DESCRIPTION
**Pull request by edzkite**
Resolves https://github.com/kiteco/kiteco/issues/12234

### CODE REVIEW

**Review by tonycheang — APPROVED** [review]
Interesting. Do we know why this is? Is this because we now wait for the copilot to open and we miss the 400 ms window `g:kite_long_timeout`? If that's the case, wondering if we should just have this method wait for longer so to not fail silently should the command be invoked and Kite's not running, since status bars are opt-in in vim. But just a minor nit.

**Comment by edzkite** [comment]
Yeah I think it's because `kite_long_timeout ` is not sufficient for this. I'll change it to something longer.

**Comment by airblade** [comment]
`g:kite_long_timeout` is also used by these asynchronous features:

- incrementing the metrics counter;
- showing docs on hover;
- producing signature completions;
- producing non-signature completions.

File: autoload/kite/client.vim [hunk 1]
```diff
@@ -132,9 +132,10 @@ endfunction
...
 
 function! kite#client#request_related(json, handler)
-  let path = s:codenav_path
+  let path = s:codenav_path 
+  let timeout = 10000 "10s
   if has('channel')
-    call s:async(function('s:timer_post', [path, g:kite_long_timeout, a:json, a:handler]))
+    call s:async(function('s:timer_post', [path, timeout, a:json, a:handler]))
   else
     call kite#async#execute(s:external_http_cmd(s:base_url.path, g:kite_long_timeout, 1),
```
Comment (airblade) [inline autoload/kite/client.vim:140]: Did you want to change the timeout for Neovim as well?

Comment (edzkite) [reply autoload/kite/client.vim:140]: Yes. How can I ensure it's also used by Neovim?

Comment (airblade) [reply autoload/kite/client.vim:140]: In the `else` block, replace `g:kite_long_timeout` with `timeout`.

**Pull request merged**